beacon-chain/blockchain/process_block.go

```diff
@@ -75,6 +75,23 @@ func (s *Service) postBlockProcess(cfg *postBlockProcessConfig) error {
 	defer reportProcessingTime(startTime)
 	defer reportAttestationInclusion(cfg.roblock.Block())
 
+	// Check for equivocation before inserting into fork choice
+	slashing, slashingErr := s.detectEquivocatingBlock(cfg.ctx, cfg.roblock)
```

Comment:
This is the wrong spot to place this detection. An equivocation will never make it to this point since we check for `s.hasSeenBlockIndexSlot` during `validateBeaconBlockPubSub`.